django/db/models/fields/related.py

```diff
@@ -294,6 +294,13 @@ def resolve_related_class(model, related, field):
                 field.do_related_class(related, model)
             lazy_related_operation(resolve_related_class, cls, self.remote_field.model, field=self)
 
+    def get_filter_kwargs_for_foreign_object(self, obj):
+        """
+        Return a dict that when passed as kwargs to self.model.filter(), yields
+        all instances that are related through this field to the foreign obj.
+        """
+        return {self.name: obj.pk}
```

Comment:
Thanks for having a look Anssi.

Can you help me understand the problem here? All this is doing (for non-generic relations, which aren't currently supported) is replacing a simple access of the ``name`` attribute ([see here](https://github.com/django/django/pull/3975/files#diff-507b415116b409afa4f723e41a759a9eL1637
)), so that generics can do something different. I don't think this can break anything that currently works.

I'll have a look at the duplication issue.